src/da14531_flasher.h

```diff
@@ -0,0 +1,46 @@
+
+#ifndef DA14531_FLASHER_H
+#define DA14531_FLASHER_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+enum flashing_state {
+    FLASHING_STATE_IDLE,
+    FLASHING_STATE_SEEN_STX,
+    FLASHING_STATE_SENT_SOH,
+    FLASHING_STATE_SEEN_SOH_ACK,
+    FLASHING_STATE_SENT_FIRMWARE,
+    FLASHING_STATE_SEND_CHKSUM_ACK,
+    FLASHING_STATE_DISABLE_UART,
+    FLASHING_STATE_ENABLE_UART,
+    FLASHING_STATE_DONE_DONE,
```

Comment:
why DONE_DONE and not DONE?

Reply:
It's a draft, at some point I had DONE and wanted one more state. I'll fix it :)